### PR TITLE
Add ToString overrides on a bunch of types to improve debugging information

### PR DESCRIPTION
**Pull request by Cheesebaron**
### Description of Change ###

Adds `ToString` overrides on some types.

This is nice especially for structs used for sensor readings.

### Bugs Fixed ###

- Related to issue #466 

### API Changes ###

None

### Behavioral Changes ###

No app behavior changes. However, debugging experience is improved.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/Battery/Battery.shared.cs [hunk 1]
```diff
@@ -100,5 +100,10 @@ internal BatteryChangedEventArgs(double level, BatteryState state, BatteryPowerS
         public BatteryState State { get; }
 
         public BatteryPowerSource PowerSource { get; }
+
+        public override string ToString()
+        {
+            return $"ChargeLevel {ChargeLevel} State {State} PowerSource {PowerSource}";
```
[the remark this answers is not in the document]
Comment (Cheesebaron) [reply Xamarin.Essentials/Battery/Battery.shared.cs]: Yep, no problem.

File: Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs [hunk 2]
```diff
@@ -99,5 +99,8 @@ internal AccelerometerData(double x, double y, double z)
 
         public override int GetHashCode() =>
             Acceleration.GetHashCode();
+
+        public override string ToString() =>
+            $"X: {Acceleration.X} Y {Acceleration.Y} Z {Acceleration.Z}";
```
[the remark this answers is not in the document]
Comment (Cheesebaron) [reply Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs]: Done

**Comment by Cheesebaron** [comment]
@jamesmontemagno I've addressed your review comments. Is there anything else you want me to do for this PR?

**Comment by jamesmontemagno** [comment]
build

**Comment by jamesmontemagno** [comment]
Looks like a few build errors for stylecop:
 Battery/Battery.shared.cs(104,45): error SA1028: Code should not contain trailing whitespace [/Users/builder/jenkins/workspace/Components-Essentials-DeviceTests-PR-Mac/Xamarin.Essentials/Xamarin.Essentials.csproj]
  Compass/Compass.shared.cs(100,45): error SA1028: Code should not contain trailing whitespace [/Users/builder/jenkins/workspace/Components-Essentials-DeviceTests-PR-Mac/Xamarin.Essentials/Xamarin.Essentials.csproj]
  Geolocation/GeolocationRequest.shared.cs(57,45): error SA1028: Code should not contain trailing whitespace [/Users/builder/jenkins/workspace/Components-Essentials-DeviceTests-PR-Mac/Xamarin.Essentials/Xamarin.Essentials.csproj]

I can fix probably unless you have time.

**Comment by Cheesebaron** [comment]
@jamesmontemagno fixed

**Comment by jamesmontemagno** [comment]
build

**Pull request merged**